test/unit/core/sagas/recommendations.ts

```diff
@@ -375,50 +375,78 @@ suite('recommendations saga', ({ expect, spy, stub }) => {
     });
 
     describe('fetchPastPurchaseNavigations()', () => {
-        // it('should generate a request when getNavigations is true', () => {
-        //   const getState = spy(() => ({ a: 1 }));
-        //   const receivePastPurchaseAllRecordCount = spy(() => 1);
-        //   const receivePastPurchaseRefinements = spy(() => 2);
-        //   const saveState = spy();
-        //   const actions = {
-        //     receivePastPurchaseAllRecordCount,
-        //     receivePastPurchaseRefinements,
-        //   };
-        //   const flux: any = { actions, saveState, store: { getState } };
-        //   const config = { b: 2 };
-        //   const result = [1, 2, 3];
-        //   const request = { c: 3 };
-        //   const transformedNav = ['f'];
-        //   const productData = { selectedNavigation: [2, 3, 5], };
-        //   const pastPurchaseSkus = [1, 2, 3, 4,];
-        //   const pastPurchasesFromSkus = { a: 'b' };
-        //   const state = { c: 'd' };
-        //   const combineNavigations = stub(SearchAdapter, 'combineNavigations').returns(productData);
-        //   const navigations = stub(PastPurchaseAdapter, 'pastPurchaseNavigations').returns(transformedNav);
-        //   const searchRequest = stub(Requests, 'search').returns(result);
-        //   stub(Tasks, 'buildRequestFromSkus').withArgs(flux, pastPurchaseSkus).returns(pastPurchasesFromSkus);
-        //   stub(RequestBuilders.pastPurchaseProductsRequest, 'composeRequest')
-        //     .withArgs(state, {
-        //       query: '',
-        //       ...pastPurchasesFromSkus,
-        //     })
-        //     .returns(request);
-        //
-        //   const task = Tasks.fetchPastPurchaseProducts(flux, true, <any>{ payload: {} });
-        //
-        //   task.next();
-        //   task.next(pastPurchaseSkus);
-        //   expect(task.next(state).value).to.eql(effects.call(searchRequest, flux, request));
-        //   expect(task.next(request).value).to.eql(effects.select(Selectors.config));
-        //   expect(task.next(config).value).to.eql(effects.put(<any>[
-        //     receivePastPurchaseAllRecordCount(),
-        //     receivePastPurchaseRefinements(),
-        //   ]));
-        //   task.next();
-        //
-        //   expect(saveState).to.not.be.called;
-        //   expect(navigations).to.be.calledWithExactly(config, productData);
-        // });
+      it('should return if there are no pastPurchaseSkus', () => {
+        const flux: any = {};
+        const action: any = {};
+        const config = {};
+
+        const task = Tasks.fetchPastPurchaseNavigations(flux, action);
+
+        expect(task.next().value).to.eql(effects.select(Selectors.pastPurchases));
+        expect(task.next([]).value).to.be.undefined;
```

Comment:
It's better to test that `.done` is `true` if you want to check that a generator has returned.